ReadMe.md
# Breeze AlphaBlack

Breeze Light theme with minor improvements and a black panel/titlebar.

https://store.kde.org/p/1084931/

## Screenshots

![](https://cdn.pling.com/img/f/1/e/1/7f50888757a024f34ef3906c0948817605b3.png)

## Install

* System Settings > Desktop Theme
* Get New Themes...
* Search for "Breeze AlphaBlack", install it, then apply it.
* If you want the breeze window decorations in black, run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/setthemecolor.py 0,0,0`

## Customize Theme

### "AlphaBlack Control" Widget

![](https://i.imgur.com/TYxCBnc.jpg)

v10 introduced an accompanying widget to easily change the theme's accent color, panel opacity, or change the taskbar to look a bit more like Windows 10.

You can [download the widget here](https://store.kde.org/p/1237963/), or install it via the GUI:

1. Right click the Panel > Add Widgets
2. Get New Widgets > Download
3. Search for "AlphaBlack Control" and install it.

After installing, the widget should appear in your system tray. If you lock your widgets, it will hide in the system tray popup.


### Command Line

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/desktoptheme.py`  
to see all commands in the help message.

#### Panel Opacity

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/desktoptheme.py set panel.opacity 0.75`  
where `0.75` means the panel is 75% visible.

You can also set the `dialog.opacity` for panel popups, and `widget.opacity` for desktop widgets.


#### Panel Color + Window Decorations

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/desktoptheme.py set theme.accentColor 255,255,255`  
where `255,255,255` is your desired RGB color.


#### Window Decorations

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/desktoptheme.py settitlebarcolor 255,255,255`  
where `255,255,255` is your desired RGB color.

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/desktoptheme.py resettitlebarcolors`  
to reapply the colors from your seleted color scheme.

#### Task Manager Theme

Run `python3 ~/.local/share/plasma/desktoptheme/breeze-alphablack/settasksvg.py outside`  
where `outside` draws the line on the edge of the screen like Windows 10, or `inside` like Breeze.



## Window Decorations

To color the the default breeze window decorations (window titlebars). You can either:

1. Download a custom window decoration.
2. Use the set titlebar colors python script above.
3. Use the System Settings > Colors > Colors section to color the titlebar background/foreground. You will not be able to color the frame color (bottom/left/right of the window) from this menu.
4. Add/set the following values in your `~/.config/kdeglobals` file to set the borders to black (when using the breeze window decorations).

```ini
[WM]
activeBackground=0,0,0
activeBlend=17,17,17
activeFont=Noto Sans,10,-1,5,50,0,0,0,0,0
activeForeground=239,240,241
inactiveBackground=8,8,8
inactiveBlend=75,71,67
inactiveForeground=189,195,199
frame=0,0,0
inactiveFrame=8,8,8
```

## Misc

* Wiki for Editing Desktop Themes  
  https://techbase.kde.org/Development/Tutorials/Plasma5/ThemeDetails
* Code that parses the `colors` file for use with the svg CSS colors.
  https://github.com/KDE/plasma-framework/blob/master/src/plasma/private/theme_p.cpp#L422


Changelog.md
## v20 - October 9 2021

* Parse `[Sub][Sections]` in kdeglobals to fix `desktoptheme.py` commands.
* Use dbus plasma script to quickly change theme faster.

## v19 - February 25 2021

* Refactor `panel-background.svg` to reuse the `dialog/background.svg`.
* Add thick margins to the panel svg to support Plasma 5.21's Margin Separator widgets. A 24px panel will have a 1px thick margin. A 30px panel will have a 4px thick margin. A 38px panel will have a 8px thick margin.

## v18 - December 11 2020

* Fix pale popups caused by the Background Contrast Desktop Effect (Issue #21). Added `ContrastEffect` properties in `metadata.desktop` in an attempt to fix it, however the proper fix was to ship a pre-generated `dialogs/background.svg`.
* Refactor `background.svg` to add `dialog.radius` which doesn't really work well.
* Fix `resettitlebarcolors` mentioned in AlphaBlackControl's Issue 2.
* Only use blue shifted DecorationFocus for Complementary colors (Issue #20)
* Set Header color group to fix white `plasmoidheading.svg` in Plasma 5.20

## v17 - June 16 2020

* Added `widgets/plasmoidheading.svg` to show the solid notification heading like Breeze. It's only displayed if the current theme has the svg, it does not inherit this svg.
* Removed all deprecated `set___color.py` scripts. Use the `desktoptheme.py` subcommands.

## v16 - March 15 2020

* Move `set___color.py` etc into `desktoptheme.py` as subcommands. Eg: `python3 desktoptheme.py set dialog.opacity 0.7`. The existing scripts are deprecated and will be removed in the next version.
* Add ability to set dialog padding with `python3 desktoptheme.py set dialog.padding 2`. Note that you need to close a popup and reopen it to see the change in padding. You may need to run the command a few times for it to take effect.

## v15 - March 17 2019

* Cleanup the desktop widget background svg coordinates (`widget/background.svg`).
* Use Breeze's rectangle new rounded corner popup dialog svg from KDE Frameworks 5.56.
* Fix the dialog popup color (when using a custom opacity) broken by the Qt 5.12.2 update. The default dialog popups and tooltips will be fixed in KDE Frameworks 5.56.2 or 5.57.0.

## v14 - December 15 2018

* Use Breeze's rectangle corner popup svg. Somewhere along the line this theme ended up using a different svg with round corners for some reason.
* Add `sethighlightcolor.py` to set the selection/highlight color.

## v13 - October 12 2018

* Hide the panel shadow when the panel opacity is set below 30% (`0.3`).
* Update the desktop widget background to use the same look as breeze.
* Add `settextcolor.py` to change the panel/widget/titlebar text color.
* Fix `resettitlebarcolor.py` not working for color schemes installed in the home directory (`~/.local/share/color-scheme/`. It previously only worked for color schemes in the root directory (`/usr/share/color-scheme/`). This fix also scans all color schemes when the color scheme filename is not the typical `{ColorSchemeName}.colors` (Eg: "Breeze Solarized Light" uses "BreezeSolarizedLight.colors").
* Add `resettodefaults.py` script to revert all changes.

## v12 - July 26 2018

* Add settitlebarcolor.py and resettitlebarcolor.py to easily reset the titlebar color. v3 of the widget has buttons that will run these scripts.

## v11 - June 24 2018

* Remove the opache and transparent folders which don't appear to do anything (they aren't overriding breeze).
* Add new templates + scripts to set the panel popups/dialog opacity, and the desktop widget opacity. The new scripts are setdialogopacity.py and setwidgetopacity.py. Both work the same as setpanelopacity.py.

## v10 - May 29 2018

* Follow color scheme when compositor is off. Gets rid of white outline when compositor is off.
* Add breeze's widgets/tabbar.svg, as Plasma doesn't allow it to "inherit" this svg. This fixes/adds the "popup is open" blue line.
* Add alternative tasks.svg where the line is on the outside/edge of the screen line Windows 10. Use the config widget, or the settasksvg.py script, to change to it.
* "Unzip" the svgz into svg, and cleanup the xml.

## v9 - February 20 2017

* Fix panel opacity script so it can be run anywhere.

## v8 - January 21 2017

* Add a script to set the panel opacity.

## v7

* Fix the minimized tasks svg (showed nothing instead of looking like a "normal" task)
* Use a hardcoded #f67400 orange for highlighted tasks instead of the NeutralText color from the color scheme.

## v6

* Use the Breeze 5.8 tasks.svg (thinner line), but with minimized tasks looking the same as a normal task.

## v5

* Add script to change the color.


## v3 - June 16 2016

* Theme progressbar in tasks.

## v2 - June 15 2016

* Got rid of the white line at the top of the panel
* Got rid of the pink spot in the pager


settextcolor.py
#!/usr/bin/python3

import sys, os
from desktoptheme import BreezeAlphaBlack

scriptFilename = 'settextcolor.py'
defaultColorName = 'white'
defaultColorStr = '239,240,241'

def printHelp():
	print("python3 {} [red],[green],[blue]".format(scriptFilename))
	print("\t[red/green/blue] are integers from 0-255")
	print("\tEg: python3 {} {}".format(scriptFilename, defaultColorStr))

if __name__ == '__main__':
	if len(sys.argv) >= 2:
		try:
			r, g, b = map(int, sys.argv[1].split(','))
			newColor = "{},{},{}".format(r, g, b)
			desktopTheme = BreezeAlphaBlack()
			desktopTheme.setTextColor(newColor)
		except Exception as e:
			print(e)
			printHelp()
			sys.exit(0)
	else:
		printHelp()
		print()
		print("No color given. Using the default {} ({}).".format(defaultColorStr, defaultColorName))
		desktopTheme = BreezeAlphaBlack()
		desktopTheme.setTextColor(defaultColorStr)
